.github/workflows/ci.yml

```diff
@@ -99,3 +104,56 @@ jobs:
       - name: Upload coverage
         if: github.repository == 'iron-fish/ironfish'
         run: CODECOV_TOKEN=${{ secrets.CODECOV_TOKEN }} ROOT_PATH=$GITHUB_WORKSPACE/ yarn coverage:upload
+
+  testperf:
+    name: Perf Tests
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check out Git repository
+        uses: actions/checkout@v3
+
+      - name: Use Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '18.12.1'
+          cache: 'yarn'
+
+      - name: Cache Rust
+        uses: Swatinem/rust-cache@v2
+        with:
+          shared-key: nodejs
+
+      - name: Install packages
+        run: yarn --non-interactive --frozen-lockfile
+
+      - name: Run perf tests
+        run: yarn test:perf:report
+
+      - name: Generate test report
+        uses: EnricoMi/publish-unit-test-result-action@v2
+        id: test-results
+        if: always()
+        with:
+          files: |
+            ironfish/test-reports/jest-junit.xml
+          job_summary: true
+          json_file: 'tests.json'
+          report_individual_runs: true
+          report_suite_logs: 'any'
+          json_suite_details: true
+          json_test_case_results: true
+      
+      - name: Output test results
+        uses: ./.github/json_output
+        if: always()
```

Comment:
Do we need these `always` statements? Will it not get executed sometimes without them?

Reply:
The step will not execute if previous step failed. In our cases, if any test fails it wont continue to run the following steps. Since these are perf tests, it will be better to run the whole flow and generate the final detailed results.